crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_none_literal.rs

```diff
@@ -185,44 +192,74 @@ fn create_fix_edit(
+            applicability,
+        )));
     }
 
-    let bin_or = Expr::BinOp(ExprBinOp {
+    let new_literal_expr = Expr::Subscript(ast::ExprSubscript {
+        value: Box::new(literal_subscript.clone()),
         range: TextRange::default(),
-        left: Box::new(Expr::Subscript(ast::ExprSubscript {
-            value: Box::new(literal_subscript.clone()),
-            range: TextRange::default(),
-            ctx: ExprContext::Load,
-            slice: Box::new(if literal_elements.len() > 1 {
-                Expr::Tuple(ast::ExprTuple {
-                    elts: literal_elements.into_iter().cloned().collect(),
-                    range: TextRange::default(),
-                    ctx: ExprContext::Load,
-                    parenthesized: true,
-                })
-            } else {
-                literal_elements[0].clone()
-            }),
-        })),
-        op: ruff_python_ast::Operator::BitOr,
-        right: Box::new(Expr::NoneLiteral(ExprNoneLiteral {
-            range: TextRange::default(),
-        })),
+        ctx: ExprContext::Load,
+        slice: Box::new(if literal_elements.len() > 1 {
+            Expr::Tuple(ast::ExprTuple {
+                elts: literal_elements.into_iter().cloned().collect(),
+                range: TextRange::default(),
+                ctx: ExprContext::Load,
+                parenthesized: true,
+            })
+        } else {
+            literal_elements[0].clone()
+        }),
     });
 
-    let content = checker.generator().expr(&bin_or);
-    Some(Edit::range_replacement(content, literal_expr.range()))
+    match union_kind {
+        UnionKind::TypingOptional => {
+            let (import_edit, bound_name) = checker.importer().get_or_import_symbol(
+                &ImportRequest::import_from("typing", "Optional"),
+                literal_expr.start(),
+                checker.semantic(),
+            )?;
+            let optional_expr = typing_optional(new_literal_expr, Name::from(bound_name));
+            let content = checker.generator().expr(&optional_expr);
+            let optional_edit = Edit::range_replacement(content, literal_expr.range());
+            Ok(Some(Fix::applicable_edits(
+                import_edit,
+                [optional_edit],
+                applicability,
+            )))
+        }
+        UnionKind::BitOr => {
+            let none_expr = Expr::NoneLiteral(ExprNoneLiteral {
+                range: TextRange::default(),
+            });
+            let union_expr = pep_604_union(&[new_literal_expr, none_expr]);
+            let content = checker.generator().expr(&union_expr);
+            let union_edit = Edit::range_replacement(content, literal_expr.range());
+            Ok(Some(Fix::applicable_edit(union_edit, applicability)))
+        }
+        // We dealt with this case earlier to avoid allocating `lhs` and `rhs`
+        UnionKind::NoUnion => {
+            unreachable!()
+        }
+    }
+}
+
+#[derive(Copy, Clone)]
```

Comment:
I tend to add these to simple enums like this, because you never know when they might be useful in tests or debug prints

```suggestion
#[derive(Copy, Clone, Debug, PartialEq, Eq)]
```